qiskit/tools/backends/reduced_cmap.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Reduced coupling_map
+"""
+
+import numpy as np
+import scipy.sparse as sp
+import scipy.sparse.csgraph as cs
+from qiskit.qiskiterror import QiskitError
+
+
+def reduced_coupling_map(backend, mapping):
+    """Returns a reduced coupling map that
+    corresponds to the subgraph of backend qubits
+    selected in the mapping.
+
+    Args:
+        backend (BaseBackend): A backend for a physical device.
+        mapping (list): A mapping of reduced qubits to device
+                        qubits.
+
+    Returns:
+        list: A reduced coupling_map for the selected qubits.
```

Comment:
the coupling map list format is useful as in input from user, but internally we use `CouplingMap` objects.

Reply:
This is for users to input into `transpile` later.  It is not meant to be used internally.